packages/collection-model/lib/model.js

```diff
@@ -270,6 +270,43 @@ const CollectionModel = AmpersandModel.extend(debounceActions(['fetch']), {
     }
   },
 
+  /**
+   * Fetches collection info and returns a special format of collection metadata
+   * that events like open-in-new-tab, select-namespace, edit-view require
+   */
+  async fetchMetadata({ dataService }) {
```

Comment:
test for this would be good, not sure how complicated that is though.

Reply:
For the purpose of avoiding further conflicts with your branch and because the whole feature is behind a feature flag I'll merge this as-is. I'll add tests in the follow-up PR as they will take some time to set up properly, yeah...